### PR TITLE
Update order status to processing on Webhook event trigger

### PR DESCRIPTION
**Pull request by nikhilsharma1100**

### CODE REVIEW

File: Controller/Payment/Webhook.php [hunk 1]
```diff
@@ -177,11 +177,14 @@ public function execute()
...
-                    $orderId = $post['payload']['payment']['entity']['notes']['merchant_order_id'];
-                    $orderWebhookData = $this->getOrderWebhookData($orderId);
+                    $orderId            = $post['payload']['payment']['entity']['notes']['merchant_order_id'];
+                    $paymentId          = $post['payload']['payment']['entity']['id'];
+                    $orderWebhookData   = $this->getOrderWebhookData($orderId);
+
                     if (empty($orderWebhookData['rzp_webhook_notified_at']) === true)
                     {
                         $this->setWebhookNotifiedAt($orderWebhookData['entity_id']);
+                        $this->setWebhookData($orderWebhookData['entity_id'], true, $paymentId, $post['event'], $post['payload']['payment']['entity']['amount']);
```
Comment (enishant) [inline Controller/Payment/Webhook.php]: @nikhilsharma1100 Paid amount has different array format as per event.
Payment Authorize: `$post['payload']['payment']['entity']['amount']`
Order Paid: `$post['payload']['order']['entity']['amount_paid']`

File: Controller/Payment/Webhook.php [hunk 2]
```diff
@@ -472,7 +475,7 @@ protected function orderPaid(array $post)
                                 $this->orderSender->send($order);
                                 $this->checkoutSession->unsRazorpayMailSentOnSuccess();
                             }
-                            catch (\Magento\Framework\Exception\MailException $exception)
+                            catch (\Magento\Framework\Exception\MailException $e)
                             {
                                 $this->logger->critical($e);
```
Comment (enishant) [inline Controller/Payment/Webhook.php]: @nikhilsharma1100 Please use `$e->getMessage()` to log exception message.

File: Controller/Payment/Webhook.php [hunk 3]
```diff
@@ -555,4 +558,19 @@ protected function setWebhookNotifiedAt($entity_id)
...
+
+    protected function setWebhookData($entityId, $webhookVerifiedStatus, $paymentId, $event, $amount)
+    {
+        $webhookData = array(
+            "webhook_verified_status"   => $webhookVerifiedStatus,
+            "payment_id"                => $paymentId,
+            "event"                     => $event,
+            "amount"                    => $amount 
+        );
+        $webhookDataText = serialize($webhookData);
```
Comment (enishant) [inline Controller/Payment/Webhook.php:581]: @nikhilsharma1100 Please add this on new line.

File: Cron/UpdateOrdersToProcessing.php [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+        foreach ($orders->getItems() as $order)
+        {
+            if ($order->getPayment()->getMethod() === 'razorpay') {
+                $rzpWebhookData = $order->getRzpWebhookData();
+                if (empty($rzpWebhookData) === false) // check if webhook cron has run and populated the rzp_webhook_data column
+                {
+                    $rzpWebhookDataObj = unserialize($rzpWebhookData);
+
+                    if($rzpWebhookDataObj['webhook_verified_status'] === true)
```
Comment (enishant) [inline Cron/UpdateOrdersToProcessing.php]: @nikhilsharma1100 Why this status check done here? 
When you are already setting this value as `true` always from `Webhook` controller.

File: Cron/UpdateOrdersToProcessing.php [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+                    "$paymentId",
+                    (new AuthorizeCommand())->execute(
+                        $payment,
+                        $order->getGrandTotal(),
+                        $order
+                    ),
+                    ""
+                );
+            }
+            else
```
Comment (enishant) [inline Cron/UpdateOrdersToProcessing.php]: @nikhilsharma1100 Please add `else if` condition for event `order.paid` here.

File: Cron/UpdateOrdersToProcessing.php [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    private function updateOrderStatus($order, $rzpWebhookData)
+    {
+        if ($order)
+        {
+            $this->logger->info("Cronjob: Updating to Processing for Order ID: " 
+                                . $order->getEntityId() 
+                                . " and Event :" 
+                                . $rzpWebhookData['event']);
+
+            $payment = $order->getPayment();
```
Comment (enishant) [inline Cron/UpdateOrdersToProcessing.php]: @nikhilsharma1100 Please update formatting to align variables with each other.

File: Cron/UpdateOrdersToProcessing.php [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+            $transaction->save();
+
+            $amountPaid = number_format($rzpOrderAmount / 100, 2, ".", "");
+
+            $order->setState(static::STATUS_PROCESSING)->setStatus(static::STATUS_PROCESSING);
+
+            $order->addStatusHistoryComment(
+                __(
+                    'Actual Amount %1 of %2, with Razorpay Offer/Fee applied.',
+                    "Authroized",
```
Comment (enishant) [inline Cron/UpdateOrdersToProcessing.php]: @nikhilsharma1100 Please include Captured comment also with condition as per event.
Reference: https://github.com/razorpay/razorpay-magento/blob/7b5f35ad08a1902facefad4794e572155e909fb7/Controller/Payment/Webhook.php#L438

File: Cron/UpdateOrdersToProcessing.php [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+                        $this->logger->critical($e);
+                    }
+                }
+            }
+
+            $order->save();
+        }
+    }
+
+    // /**
```
Comment (enishant) [inline Cron/UpdateOrdersToProcessing.php]: @nikhilsharma1100 Please remove unused commented code

**Review by enishant — CHANGES_REQUESTED** [review]
@nikhilsharma1100 Please update commented changes with this PR

File: Cron/UpdateOrdersToProcessing.php (hunk 4 above)
Comment (nikhilsharma1100) [reply Cron/UpdateOrdersToProcessing.php]: Yes correct, will remove the condition

File: Cron/UpdateOrdersToProcessing.php (hunk 7 above)
Comment (nikhilsharma1100) [reply Cron/UpdateOrdersToProcessing.php]: okay, will do

File: Controller/Payment/Webhook.php (hunk 2 above)
Comment (nikhilsharma1100) [reply Controller/Payment/Webhook.php]: okay

File: Controller/Payment/Webhook.php (hunk 3 above)
Comment (nikhilsharma1100) [reply Controller/Payment/Webhook.php:581]: okay

File: Cron/UpdateOrdersToProcessing.php (hunk 6 above)
Comment (nikhilsharma1100) [reply Cron/UpdateOrdersToProcessing.php]: done

File: Cron/UpdateOrdersToProcessing.php (hunk 5 above)
Comment (nikhilsharma1100) [reply Cron/UpdateOrdersToProcessing.php]: done

File: Cron/UpdateOrdersToProcessing.php (hunk 8 above)
Comment (nikhilsharma1100) [reply Cron/UpdateOrdersToProcessing.php]: okay

**Review by sofiya02 — APPROVED** [review]
LGTM

File: Controller/Payment/Webhook.php (hunk 1 above)
Comment (nikhilsharma1100) [reply Controller/Payment/Webhook.php]: okay, assigning value based on event

**Pull request merged**